sbin/build.sh

```diff
@@ -22,211 +22,260 @@ OPENJDK_REPO_NAME=$3
-
-if [[ -f /.dockerenv ]] ; then
-  echo "Detected we're in docker"
-  WORKING_DIR=/openjdk/jdk8u
-  TARGET_DIR=$WORKING_DIR
-fi
-
-# E.g. /openjdk/jdk8u if you're building in a Docker container
-# otherwise ensure it's a writable area e.g. /home/youruser/myopenjdkarea
-
-if [ -z "$WORKING_DIR" ] || [ -z "$TARGET_DIR" ] ; then
-    echo "build.sh is called by makejdk.sh and requires two parameters"
-    echo "Are you sure you want to call it directly?"
-    echo "Usage: bash ./build.sh <workingarea> <targetforjdk>"
-    echo "Note that you must have the OpenJDK source before using this script!"
-    echo "This script will try to move ./openjdk to the source directory for you, "
-    echo "and this will be your working area where all required files will be downloaded to."
-    echo "You can override the JDK boot directory by setting the environment variable JDK_BOOT_DIR"
-    exit;
-fi
-
-echo "Making the working directory to store source files and extensions: $WORKING_DIR"
-
-mkdir -p $WORKING_DIR
-
-cd $WORKING_DIR
+initialiseEscapeCodes()
+{
+  # Escape code
+  esc=$(echo -en "\033")
+
+  # Set colors
+  error="${esc}[0;31m"
+  good="${esc}[0;32m"
+  info="${esc}[0;33m"
+  git="${esc}[0;34m"
+  normal=$(echo -en "${esc}[m\017")
+}
+
+checkIfDockerIsUsedForBuildingOrNot()
+{
+  # If on docker
+
+  if [[ -f /.dockerenv ]] ; then
+    echo "Detected we're in docker"
+    WORKING_DIR=/openjdk/jdk8u
+    TARGET_DIR=$WORKING_DIR
+  fi
 
-############################################
+  # E.g. /openjdk/jdk8u if you're building in a Docker container
+  # otherwise ensure it's a writable area e.g. /home/youruser/myopenjdkarea
+
+  if [ -z "$WORKING_DIR" ] || [ -z "$TARGET_DIR" ] ; then
+      echo "build.sh is called by makejdk.sh and requires two parameters"
+      echo "Are you sure you want to call it directly?"
+      echo "Usage: bash ./build.sh <workingarea> <targetforjdk>"
+      echo "Note that you must have the OpenJDK source before using this script!"
+      echo "This script will try to move ./openjdk to the source directory for you, "
+      echo "and this will be your working area where all required files will be downloaded to."
+      echo "You can override the JDK boot directory by setting the environment variable JDK_BOOT_DIR"
+      exit;
+  fi
 
-echo "Downloading required dependencies..."
+  echo "Making the working directory to store source files and extensions: $WORKING_DIR"
 
-# ALSA first for sound
+  mkdir -p $WORKING_DIR
 
-echo "Checking for ALSA"
+  cd $WORKING_DIR
+}
 
-FOUND_ALSA=$(find "$WORKING_DIR" -name "alsa-lib-1.0.27.2")
+checkingAndDownloadingAsla()
+{
+  # ALSA first for sound
 
-if [[ ! -z $FOUND_ALSA ]] ; then
-  echo "Skipping ALSA download"
-else
-  wget -nc ftp://ftp.alsa-project.org/pub/lib/alsa-lib-1.0.27.2.tar.bz2
-  tar xvf alsa-lib-1.0.27.2.tar.bz2
-  rm alsa-lib-1.0.27.2.tar.bz2
-fi
+  echo "Checking for ALSA"
 
-echo "Checking for freetype"
+  FOUND_ALSA=$(find "$WORKING_DIR" -name "alsa-lib-1.0.27.2")
```

Comment:
Wasn't the primary purpose when I was extracting the code blocks, I can extract all these kinds of dependencies in an PR